### PR TITLE
Add a basic sort function for RenderInstancedQueue.

### PR DESCRIPTION
**Pull request by yiwenxue**
Re: #https://github.com/cocos/3d-tasks/issues/12156, https://github.com/cocos/3d-tasks/issues/5937

RenderInstanced doesn't have the `sort` function to manage the drawing order, and the drawing order only depends on the camera depth of drawing objects. So the transparent object may be drawn before the opaque object. 

This PR adds a basic sort function to separate the opaque and solid objects. With `std::stable_partation()` the original order of opaque objects and the order of transparent objects will keep unchanged.

### Changelog

* add a basic sort function to reorder the instanced queues.

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check // Manual trigger with `@cocos-robot run test cases` afterward

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by yiwenxue** [comment]
@cocos-robot run test cases

File: native/cocos/renderer/pipeline/RenderInstancedQueue.cpp [hunk 1]
```diff
@@ -35,9 +35,18 @@ void RenderInstancedQueue::clear() {
     for (auto *it : _queues) {
         it->clear();
     }
+    _renderQueues.clear();
     _queues.clear();
 }
 
+void RenderInstancedQueue::sort() {
+    std::copy(_queues.cbegin(), _queues.cend(), std::back_inserter(_renderQueues));
+    auto isSolid = [](const InstancedBuffer *instance) {
```
Comment (star-e) [inline native/cocos/renderer/pipeline/RenderInstancedQueue.cpp]: We use "opaque" to describe object which is not transparent/cutout.

**Comment by yiwenxue** [comment]
@cocos-robot run test cases

**Comment by yiwenxue** [comment]
https://github.com/cocos/3d-tasks/issues/5937
https://github.com/cocos/3d-tasks/issues/12155

**Comment by yiwenxue** [comment]
@cocos-robot run test cases

**Comment by yiwenxue** [comment]
@cocos-robot run test cases

**Comment by yiwenxue** [comment]
@cocos-robot run test cases

**Comment by yiwenxue** [comment]
@cocos-robot run test cases

**Comment by yiwenxue** [comment]
@cocos-robot run test cases

**Comment by yiwenxue** [comment]
@cocos-robot run test cases

**Comment by yiwenxue** [comment]
@cocos-robot run test cases

**Comment by yiwenxue** [comment]
@cocos-robot run test cases

**Comment by yiwenxue** [comment]
@cocos-robot run test cases

**Comment by star-e** [comment]
@cocos-robot run test cases

**Comment by star-e** [comment]
https://forum.cocos.org/t/topic/135885/2
issue on forum

**Comment by star-e** [comment]
The failure cases on windows are related to shadowmap and animation.

**Review by star-e — APPROVED** [review]

**Pull request merged**